service/src/main/java/bio/terra/profile/app/common/MetricUtils.java

```diff
@@ -0,0 +1,44 @@
+package bio.terra.profile.app.common;
+
+import bio.terra.profile.model.CloudPlatform;
+import bio.terra.profile.service.profile.model.BillingProfile;
+import io.micrometer.core.instrument.Metrics;
+import java.util.concurrent.Callable;
+
+public class MetricUtils {
+
+  private static final String CLOUD_PLATFORM_TAG = "cloudPlatform";
+
+  /**
+   * Emit a metric for the duration of creating a billing profile.
+   *
+   * @param callable the code to execute to create the profile, which will be timed
+   * @param platform the platform the profile will be created on
+   */
+  public static BillingProfile recordProfileCreation(
+      Callable<BillingProfile> callable, CloudPlatform platform) {
+    try {
+      return Metrics.globalRegistry
+          .timer("profile.creation.time", CLOUD_PLATFORM_TAG, platform.toString())
+          .recordCallable(callable);
+    } catch (Exception ex) { // Any exception bubbling up from the callable.
+      // The method we are calling does not declare checked exceptions, so retain the original.
```

Comment:
This is ugly,  but the recordCallable method declares that it throws Exception (to bubble up anything from the callable). The actual method we are calling (submitAndWait) does not declare any checked exceptions. I want to retain the originally thrown exception for clarity, so doing this casting.